service/worker/batcher/workflow.go

```diff
@@ -20,7 +20,332 @@
 
 package batcher
 
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/log"
+	"github.com/uber/cadence/common/log/tag"
+	"go.uber.org/cadence"
+	"go.uber.org/cadence/.gen/go/shared"
+	"go.uber.org/cadence/activity"
+	"go.uber.org/cadence/workflow"
+	"golang.org/x/time/rate"
+)
+
 const (
 	batcherContextKey   = "batcherContext"
 	batcherTaskListName = "cadence-sys-batcher-tasklist"
+	batchWFTypeName     = "cadence-sys-batch-workflow"
+	batchActivityName   = "cadence-sys-batch-activity"
+
+	infiniteDuration = 20 * 365 * 24 * time.Hour
+	pageSize         = 1000
+
+	// below are default values for BatchParams
+
+	defaultRPS                      = 50
+	defaultConcurrency              = 5
+	defaultAttemptsOnRetryableError = 50
+	defaultActivityHeartBeatTimeout = time.Minute
+)
+
+const (
+	BatchTypeTerminate = "terminate"
+	BatchTypeReset     = "reset"
+)
+
+type (
+	// BatchParams is the parameters for batch operation workflow
+	BatchParams struct {
+		// Target domain to execute batch operation
+		DomainName string
+		// To get the target workflows for processing
```

Comment:
This comment will fail lint.

Reply:
```
longer@~/gocode/src/github.com/uber/cadence:(batch-term)$ make lint
./install-dep.sh
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/cask).
No changes to formulae.

Warning: dep 0.5.4 is already installed and up-to-date
To reinstall 0.5.4, run `brew reinstall dep`
dep ensure
Running linter
```
This is OK because the upper level(BatchParams) already satisfy Linter's requirement